clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1507,7 +1515,8 @@ scheduler_tmpl_t<RecordType, ReaderType>::get_initial_input_content(
                 }
                 if (record_type_is_timestamp(record, input.next_timestamp))
                     found_timestamp = true;
-                if (found_filetype && found_timestamp)
+                if (!process_next_initial_record(input, record, found_filetype,
```

Comment:
nit: `if (!found_filetype || !found_timestamp) {` at lines 1502 and 1525 check the same condition as the "default" (non-overidden) `process_next_initial_record()`.
I assume this was intentional (perhaps we don't want subclasses to change those two checks), but I thought I'd point it out, in case we actually want them to be the same.

Reply:
Good point: it does seem it would be best to put the call to the new function in both places.  It is unlikely there is anything in the queue so in practice that part generally won't matter; the up-front check could matter if timestamp deps are turned off.  We have to pass an invalid record at those points but that is fine (added note about that to the decl).  Done.